### PR TITLE
Add Namespace to Secret type

### PR DESCRIPTION
**Pull request by jonatasbaldin**
Signed-off-by: Jonatas Baldin <jonatas.baldin@gmail.com>

## Description
Enables Namespace in Secret. Depends on [faas-provider #28](https://github.com/openfaas/faas-provider/pull/28).

## Motivation and Context
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md)) [faas-netes #511](https://github.com/openfaas/faas-netes/issues/511).
- [x] My issue has received approval from the maintainers or lead with the `design/approved` label

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: gateway/requests/requests.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type DeleteFunctionRequest struct {
 
 // Secret for underlying orchestrator
 type Secret struct {
-	Name  string `json:"name"`
-	Value string `json:"value,omitempty"`
+	Name      string `json:"name"`
```
Comment (alexellis) [inline gateway/requests/requests.go]: Why are we using the Secret struct from faas, if we have it in faas-provider too? Can we delete this copy?

Comment (jonatasbaldin) [reply gateway/requests/requests.go]: Yeah, took a look here and it's not being used anywhere. Removing it.

**Comment by jonatasbaldin** [comment]
Can I update the `faas-provider` version already?

**Comment by alexellis** [comment]
Sure, I cut a new release for you.

**Pull request merged**